src/test/java/org/apache/accumulo/proxy/its/SimpleProxyBase.java

```diff
@@ -974,320 +975,232 @@ public void testNamespaceClassLoadLoginFailure() {
-    } catch (NamespaceNotFoundException ex) {}
-    try {
-      client.removeNamespaceProperty(creds, doesNotExist, "table.compaction.major.ratio");
-      fail("exception not thrown");
-    } catch (NamespaceNotFoundException ex) {}
-    try {
-      client.getNamespaceProperties(creds, doesNotExist);
-      fail("exception not thrown");
-    } catch (NamespaceNotFoundException ex) {}
-    try {
-      IteratorSetting setting = new IteratorSetting(100, "DebugTheThings",
-          DebugIterator.class.getName(), Collections.emptyMap());
-      client.attachNamespaceIterator(creds, doesNotExist, setting,
-          EnumSet.allOf(IteratorScope.class));
-      fail("exception not thrown");
-    } catch (NamespaceNotFoundException ex) {}
-    try {
-      client.removeNamespaceIterator(creds, doesNotExist, "DebugTheThings",
-          EnumSet.allOf(IteratorScope.class));
-      fail("exception not thrown");
-    } catch (NamespaceNotFoundException ex) {}
-    try {
-      client.getNamespaceIteratorSetting(creds, doesNotExist, "DebugTheThings", IteratorScope.SCAN);
-      fail("exception not thrown");
-    } catch (NamespaceNotFoundException ex) {}
-    try {
-      client.listNamespaceIterators(creds, doesNotExist);
-      fail("exception not thrown");
-    } catch (NamespaceNotFoundException ex) {}
-    try {
-      IteratorSetting setting = new IteratorSetting(100, "DebugTheThings",
-          DebugIterator.class.getName(), Collections.emptyMap());
-      client.checkNamespaceIteratorConflicts(creds, doesNotExist, setting,
-          EnumSet.allOf(IteratorScope.class));
-      fail("exception not thrown");
-    } catch (NamespaceNotFoundException ex) {}
-    try {
-      client.addNamespaceConstraint(creds, doesNotExist, MaxMutationSize.class.getName());
-      fail("exception not thrown");
-    } catch (NamespaceNotFoundException ex) {}
-    try {
-      client.removeNamespaceConstraint(creds, doesNotExist, 1);
-      fail("exception not thrown");
-    } catch (NamespaceNotFoundException ex) {}
-    try {
-      client.listNamespaceConstraints(creds, doesNotExist);
-      fail("exception not thrown");
-    } catch (NamespaceNotFoundException ex) {}
-    try {
-      client.testNamespaceClassLoad(creds, doesNotExist, DebugIterator.class.getName(),
-          SortedKeyValueIterator.class.getName());
-      fail("exception not thrown");
-    } catch (NamespaceNotFoundException ex) {}
+    IteratorSetting iteratorSetting = new IteratorSetting(100, "DebugTheThings",
+        DebugIterator.class.getName(), Collections.emptyMap());
+
+    assertThrows(NamespaceNotFoundException.class,
+        () -> client.deleteNamespace(creds, doesNotExist));
+
+    assertThrows(NamespaceNotFoundException.class,
+        () -> client.renameNamespace(creds, doesNotExist, "abcdefg"));
+
+    assertThrows(NamespaceNotFoundException.class, () -> client.setNamespaceProperty(creds,
+        doesNotExist, "table.compaction.major.ratio", "4"));
```

Comment:
Since a lot of these are checking for the same exception, I was thinking you could just loop over all these lambda's, something like:

```java
    Stream.<Executable>of(() -> client.deleteNamespace(creds, doesNotExist),
      () -> client.renameNamespace(creds, doesNotExist, "abcdefg"),
      ...

    .forEach(executable -> assertThrows(NamespaceNotFoundException.class, executable));
 ```

This removes a lot of boilerplate, since we're testing the same thing over and over again. This is great if everything is passing, but it might be hard to troubleshoot which one had a problem when there's a failure. That might be easier to troubleshoot by adding a loop counter to track which executable failed to throw the expected exception and broke out of the loop.

In any case, it's fine the way you have it now.

Reply:
I was considering doing something like this. I'll take another look

Reply:
Made these changes to one of the tests in e2fb8c55e1537194de54fdf8c35bd647a5eda5f4. Working on the others now.